### ISSUE TITLE
Support for Caps Lock and Backspace

### ISSUE THREAD

**Issue by MuseHorizon** [phil294/AHK_X11#36]
Hi Phil,

Request to please add support for backspace and caps lock.

Example script below that turns caps lock into backspace and shift + caps lock into caps lock:

```ahk
+CapsLock::CapsLock
CapsLock::Backspace
```

Running this returns 'capslock' not found and 'backspace' not found.

Thanks!

**Comment by phil294** [comment]
Hi!

this is mostly supported already. What you want is

```ahk
CapsLock::Send, {BackSpace}
```
Explanation:
- The `CapsLock::Backspace` is "key remapping" syntax, not supported by Windows AHK v1.0.24 and thus not supported by AHK_X11. More details about this state of the art is in the docs under "Remapping Keyboard, Joystick and Mouse" (crossed out) @ https://phil294.github.io/AHK_X11/#Remap.htm. You could theoretically put something together with `KeyWait` but this usually doesn't work yet because of internal to do stuff.
- In other words, `capslock::send {bs}` only works for single key down press, not for keeping it pressed down. That's unfortunate but the only way right now and for quite some time. If you *really* need an actual key remapping, you probably need to use xkb tools https://wiki.archlinux.org/title/Xorg/Keyboard_configuration or maybe another userspace tool exists that can do that, I don't know
- In your code, line 1, `+CapsLock::CapsLock` should never be necessary because in line 2, you're only remapping `CapsLock` but without the `*`, so shift+caps preserves its natural function anyway. If you try typing shift+caps without any ahk script running you'll notice that it already properly toggles capslock.
- When it says `capslock not found`, it actually refers to the right hand side of the `::` because CapsLock is a key, not a command. The key itself (left side of `::`) is understood properly, I added that key [just recently](https://github.com/phil294/AHK_X11/issues/35)
- CapsLock hotkey/sending doesn't work particularly great, at least on my system. If you're experimenting with this I suggest you try other "normal" keys first to get a feeling for it.

TL;DR: Try my suggestion above, if it doesn't suit you, I don't think there's much to be done about it

**Comment by MuseHorizon** [comment]
Thanks for the detailed explanation, appreciate it!
